voila/handler.py

```diff
@@ -67,6 +67,32 @@ def get(self, path=None):
             'theme': self.voila_configuration.theme
         }
 
+        # add extra resources (necessary for reveal template)
+        resources_reveal = {
+            'reveal': {
+                'theme': 'simple',
+                'transition': 'slide',
+                'scroll': False,
+            }
+        }
+        resources.update(resources_reveal)
+
+        # include potential extra resources
+        extra_resources = self.voila_configuration.resources
+        if extra_resources:
+            for k in extra_resources.keys():
+                if k in resources.keys():
+                    if isinstance(extra_resources[k], dict):
+                        for kk in extra_resources[k].keys():
+                            # update or add depth-2 resources
+                            resources[k][kk] = extra_resources[k][kk]
+                        else:
+                            # update depth-1 (top-level) resources
+                            resources[k] = extra_resources[k]
+                else:
+                    # add depth-1 (top-level) resources
+                    resources[k] = extra_resources[k]
```

Comment:
Maybe we can do a sort of recursive extend.

From the SO answer here: 
https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth

Python 2

```python
import collections

def update(d, u):
    for k, v in u.iteritems():
        if isinstance(v, collections.Mapping):
            d[k] = update(d.get(k, {}), v)
        else:
            d[k] = v
    return d
```

Python 3:

```python
import collections

def update(d, u):
    for k, v in u.items():
        if isinstance(v, collections.Mapping):
            d[k] = update(d.get(k, {}), v)
        else:
            d[k] = v
    return d
```

Reply:
Thanks, @SylvainCorlay !

I have named this utility function `update_nested_dict()` to be more explicit and not to step on `dict.update()`'s toes. I guess we don't need to support Python 2... am I right?

Reply:
We still have py2 support, but your last commit fixed this.